source/Rectangle.cpp

```diff
@@ -187,3 +191,15 @@ bool Rectangle::Overlaps(const Rectangle &other) const
 {
 	return !(other.Left() > Right() || other.Right() < Left() || other.Top() > Bottom() || other.Bottom() < Top());
 }
+
+
+
+bool Rectangle::Overlaps(const Point &circle, const double radius) const
+{
+	// Handle case where circle is entirely within rectangle.
+	if(Contains(circle))
+		return true;
+
+	const Point closest = {max(Left(), min(Right(), circle.X())), max(Top(), min(Bottom(), circle.Y()))};
+	return (circle - closest).LengthSquared() < radius * radius;
```

Comment:
[That was fun to verify for myself.](https://www.desmos.com/calculator/bsr8eeykvm)
```suggestion
	const Point closest = Point(max(Left(), min(Right(), circle.X())), max(Top(), min(Bottom(), circle.Y())));
	return (circle - closest).LengthSquared() < radius * radius;
```